src/tracers.jl

```diff
@@ -1,23 +1,37 @@
 abstract type AbstractTracer <: Real end
 
-myempty(::T) where {T<:AbstractTracer} = myempty(T)
-myempty(::Type{T}) where {T<:AbstractTracer} = T()
+sparse_vector(T, index::Integer) = T([index])
+
+#===================#
+# Set operations    #
+#===================#
 
 myempty(::S) where {S<:AbstractSet} = myempty(S)
 myempty(::Type{S}) where {S<:AbstractSet} = S()
 
-sparse_vector(T, index::Integer) = T([index])
+function clever_union(a::AbstractSet{I}, b::AbstractSet{I}) where {I}
+    if isempty(a)
+        return b
+    elseif isempty(b)
+        return a
+    else
+        return union(a, b)
+    end
+end
```

Comment:
Will be done in my PR on #80.

Reply:
I think it needs to be handled at the level of sets, not tracers. Especially for things like the future `TapeSet`

Reply:
Just saying that in the future, I'd prefer for PRs not to implement unrelated features that are assigned to other people in other issues.

But now that it's done, let's merge this and leave #80 open.

Reply:
I will revert this, it just seemed natural when I wrote it

Reply:
Keep it in!